File: examples/python/hello.py
#!/usr/bin/env python3

"""
Example Python application using Claude Code SDK with OAuth authentication
This demonstrates how to use Claude Code in a Docker container
"""

import os
import asyncio
from claude_code_sdk import query, ClaudeCodeOptions
from claude_code_sdk.types import AssistantMessage, TextBlock

async def example_usage():
    print("🤖 Claude Code SDK Example (Python)")
    print("====================================\n")
    
    # Check authentication
    oauth_token = os.getenv('CLAUDE_CODE_OAUTH_TOKEN')
    session_token = os.getenv('CLAUDE_CODE_SESSION')
    
    if not oauth_token and not session_token:
        print("❌ No authentication token found!")
        print("Please set CLAUDE_CODE_OAUTH_TOKEN environment variable")
        return False
    
    print("✓ Authentication token found")
    if oauth_token:
        print(f"  OAuth token: {oauth_token[:20]}...")
    if session_token:
        print(f"  Session token: {session_token[:20]}...")
    
    try:
        # Configure Claude options
        options = ClaudeCodeOptions(
            model='claude-3-5-sonnet-20241022',
            # You can add other options here:
            # max_turns=1,
            # system_prompt='You are a helpful assistant',
            # allowed_tools=['Read', 'Write', 'Bash'],
        )
        
        prompt = """
        Hello Claude! Please help me understand how to use the Claude Code SDK.
        Can you give me a brief overview of what it can do?
        """
        
        print("📤 Sending prompt to Claude...\n")
        print("💬 Claude's Response:")
        print("─" * 50)
        
        # Stream the response
        response_content = ""
        async for message in query(prompt=prompt.strip(), options=options):
            if isinstance(message, AssistantMessage):
                for block in message.content:
                    if isinstance(block, TextBlock):
                        content = block.text
                        print(content, end="", flush=True)
                        response_content += content
        
        print("\n" + "─" * 50)
        print("✅ Response complete!")
        print(f"📊 Total characters received: {len(response_content)}")
        return True
        
    except Exception as error:
        print(f"❌ Error: {error}")
        return False

async def main():
    try:
        success = await example_usage()
        if not success:
            exit(1)
    except Exception as error:
        print(f"💥 Fatal error: {error}")
        exit(1)

if __name__ == "__main__":
    asyncio.run(main())

File: examples/javascript/hello.js
#!/usr/bin/env node

/**
 * Example JavaScript application using Claude Agent SDK with OAuth authentication
 * This demonstrates how to use Claude Agent SDK in a Docker container
 */

const { query } = require("@anthropic-ai/claude-agent-sdk");

async function exampleUsage() {
    console.log("🤖 Claude Agent SDK Example (JavaScript)");
    console.log("========================================\n");
    
    // Check authentication
    const oauthToken = process.env.CLAUDE_CODE_OAUTH_TOKEN;
    const sessionToken = process.env.CLAUDE_CODE_SESSION;
    
    if (!oauthToken && !sessionToken) {
        console.error("❌ No authentication token found!");
        console.log("Please set CLAUDE_CODE_OAUTH_TOKEN environment variable");
        return false;
    }
    
    console.log("✓ Authentication token found");
    if (oauthToken) {
        console.log(`  OAuth token: ${oauthToken.substring(0, 20)}...`);
    }
    if (sessionToken) {
        console.log(`  Session token: ${sessionToken.substring(0, 20)}...`);
    }
    
    try {
        const prompt = `
        Hello Claude! Please help me understand how to use the Claude Agent SDK.
        Can you give me a brief overview of what it can do?
        `;
        
        console.log("📤 Sending prompt to Claude...\n");
        console.log("💬 Claude's Response:");
        console.log("─".repeat(50));
        
        // Stream the response
        let responseContent = "";
        const messages = [];
        
        for await (const message of query({
            prompt: prompt.trim(),
            abortController: new AbortController(),
            options: {
                model: 'claude-3-5-sonnet-20241022',
                // You can add other options here:
                // maxTurns: 1,
                // systemPrompt: 'You are a helpful assistant',
                // allowedTools: ['Read', 'Write', 'Bash'],
            }
        })) {
            messages.push(message);
            
            // Extract and display text content from assistant messages
            if (message.type === 'assistant' && message.message?.content) {
                for (const block of message.message.content) {
                    if (block.type === 'text') {
                        const content = block.text;
                        process.stdout.write(content);
                        responseContent += content;
                    }
                }
            }
        }
        
        console.log("\n" + "─".repeat(50));
        console.log("✅ Response complete!");
        console.log(`📊 Total characters received: ${responseContent.length}`);
        console.log(`📋 Messages processed: ${messages.length}`);
        return true;
        
    } catch (error) {
        console.error(`❌ Error: ${error.message || error}`);
        return false;
    }
}

async function main() {
    try {
        const success = await exampleUsage();
        if (!success) {
            process.exit(1);
        }
    } catch (error) {
        console.error(`💥 Fatal error: ${error.message || error}`);
        process.exit(1);
    }
}

// Run the example
if (require.main === module) {
    main();
}